pkg/alibabacloud/api/api.go

```diff
@@ -396,32 +396,29 @@ func (c *Client) UnassignPrivateIPAddresses(ctx context.Context, eniID string, a
 	return err
 }
 
-func (c *Client) describeNetworkInterfaces(ctx context.Context, subnets ipamTypes.SubnetMap) ([]ecs.NetworkInterfaceSet, error) {
+func (c *Client) describeNetworkInterfaces(ctx context.Context) ([]ecs.NetworkInterfaceSet, error) {
 	var result []ecs.NetworkInterfaceSet
 
-	for _, subnet := range subnets {
-		for i := 1; ; {
-			req := ecs.CreateDescribeNetworkInterfacesRequest()
-			req.PageNumber = requests.NewInteger(i)
-			req.PageSize = requests.NewInteger(50)
-			req.VSwitchId = subnet.ID
-			c.limiter.Limit(ctx, "DescribeNetworkInterfaces")
-			resp, err := c.ecsClient.DescribeNetworkInterfaces(req)
-			if err != nil {
-				return nil, err
-			}
-			if len(resp.NetworkInterfaceSets.NetworkInterfaceSet) == 0 {
-				break
-			}
+	for i := 1; ; {
```

Comment:
suggest use NextToken and MaxResults instead of pageNumber .see https://next.api.aliyun.com/api/Ecs/2014-05-26/DescribeNetworkInterfaces?sdkStyle=old&tab=DOC

Reply:
Thanks for the info! I'll test the performance using `MaxResults` internally first.